packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -238,9 +238,162 @@ typedef bool RoutePredicate(Route<dynamic> route);
+/// called "screens" or "pages". In Flutter these elements are called
+/// routes and they're managed by a [Navigator] widget. The navigator
+/// manages a stack of [Route] objects and provides methods for managing
+/// the stack, like [Navigator.push] and [Navigator.pop].
+///
+/// ### Displaying a full-screen route
+///
+/// Although you can create a navigator directly, it's most common to use
+/// the navigator created by a [WidgetsApp] or a [MaterialApp] widget. You
+/// can refer to that navigator with [Navigator.of].
+///
+/// A MaterialApp is the simplest way to set things up. The MaterialApp's
+/// home becomes the route at the bottom of the Navigator's stack. It is
+/// what you see when the app is launched.
+///
+/// ```dart
+/// void main() {
+///   runApp(new MaterialApp(home: new MyAppHome()));
+/// }
+/// ```
+///
+/// To push a new route on the stack you can create an instance of
+/// [MaterialPageRoute] with a builder function that creates whatever you
+/// want to appear on the screen. For example:
+///
+/// ```dart
+/// Navigator.of(context).push(new MaterialPageRoute(
+///   builder: (BuildContext context) {
+///     return new Scaffold(
+///       appBar: new AppBar(title: new Text('My Page')),
+///       body: new Center(
+///         child: new FlatButton(
+///           child: new Text('POP'),
+///           onPressed: () {
+///             Navigator.of(context).pop();
+///           },
+///         ),
+///       ),
+///     );
+///   },
+/// ));
+/// ```
+///
+/// The route defines its widget with a builder function instead of a
+/// child widget because it will be built and rebuilt in different
+/// contexts depending on when it's pushed and popped.
+///
+/// As you can see, the new route can be popped, revealing the app's home
+/// page, with the Navigator's pop method:
+///
+/// ```dart
+/// Navigator.of(context).pop();
+/// ```
+///
+/// It usually isn't necessary to provide a widget that pops the Navigator
+/// in a route with a Scaffold because the Scaffold automatically adds a
+/// 'back' button to its AppBar. Pressing the back button causes
+/// [Navigator.pop] to be called. On Android, pressing the system back
+/// button does the same thing.
+///
+/// ### Using named navigator routes
+///
+/// Mobile apps often manage a large number of routes and it's often
+/// easiest to refer to them by name. The [MaterialApp] can be created
+/// with a `Map<String, WidgetBuilder>` which maps from a route's name to
+/// a builder function that will create it. The [MaterialApp] uses this
+/// map to create a value for its navigator's [onGenerateRoute] callback.
+///
+/// ```dart
+/// void main() {
+///   runApp(new MaterialApp(
+///     home: new MyAppHome(),
+///     routes: <String, WidgetBuilder> {
+///       'A': (BuildContext context) => new MyPage(title: 'A'),
+///       'B': (BuildContext context) => new MyPage(title: 'B'),
+///       'C': (BuildContext context) => new MyPage(title: 'C'),
```

Comment:
I'd probably give these slightly different titles so they don't match the route name.  Also, I think we typically start routes with `/`, so maybe the routes should be called `/a`, `/b`, etc.